module/src/components/singleSelect/singleSelect.component.tsx

```diff
@@ -0,0 +1,147 @@
+import { Form } from "../../hooks";
+import { IIcon, IconSet } from "../icon";
+import { ValidationErrors } from "../validationErrors";
+
+export type ReactSingleSelectRef = SelectRef<
+  ISelectOptionType<string>,
+  false,
+  GroupBase<ISelectOptionType<string>>
+>;
+
+export type ISelectOptionType<TSelectData = any> = {
+  value: TSelectData;
+  label: string;
+};
+
+export interface ISingleSelectProps<TSelectData = string> {
+  /**  prop for binding to an Armstrong form binder (see forms documentation) */
+  bind: Form.IBindingProps<string>;
+
+  /** the options to be displayed in the input */
+  options: ISelectOptionType<TSelectData>[];
+
+  /** adds a label above the input */
+  label?: string;
+
+  /** overrides the aria-label of the input. This is set as default to 'single-select-input' */
+  ariaLabel?: string;
+
+  /** overrides the placeholder string in the input when nothing is selected. This is set as default to 'Please select... */
+  placeholder?: string;
+
+  /** overrides the error messaging and icon display used in the error validation display */
+  validationMode?: "icon" | "message" | "both";
+
+  /** overrides the error message(s) used in the validation display */
+  errorMessages?: Form.ValidationMessage[];
+
+  /** overrides the icon option used beside the validation display */
+  errorIcon?: IIcon<IconSet>;
+
+  /** overrides the handleChange method used when the input option is changed */
+  onSelectOption?: (newValue: OnChangeValue<unknown, false>) => void;
+}
+
+import "./singleSelect.basic.scss";
+
+export const SingleSelect = React.forwardRef<
+  ReactSingleSelectRef,
+  ISingleSelectProps
+>(
+  (
+    {
+      bind,
+      options,
+      label,
+      placeholder,
+      validationMode,
+      errorMessages,
+      errorIcon,
+      ariaLabel,
+      onSelectOption,
+    },
+    ref
+  ) => {
+    const id = React.useId();
+
+    const [value, setValue, config] = Form.useBindingState(bind, {
+      validationErrorMessages: errorMessages,
+      validationErrorIcon: errorIcon,
+      validationMode,
+      onChange: onSelectOption,
+    });
+
+    const {
+      validationErrorMessages,
+      shouldShowValidationErrorMessage,
+      shouldShowValidationErrorIcon,
+      validationErrorIcon,
+    } = config;
+
+    const handleChange = React.useCallback(
+      (newValue: OnChangeValue<unknown, false>) => {
+        const castValue = newValue as ISelectOptionType<string>;
+        setValue?.(castValue?.value ?? undefined);
+      },
+      [setValue]
+    );
+
+    const selectedValue = React.useMemo(() => {
+      return options?.find(
+        (option) => option.value === (value ?? "")
+      ) as ISelectOptionType<string>;
+    }, [options, value]);
+
+    const getOptionLabel = (option?: ISelectOptionType<string>): string => {
+      return option?.label ?? "";
+    };
+
+    const getOptionValue = (option?: ISelectOptionType<string>): string => {
+      return option?.value ?? "";
+    };
```

Comment:
I don't think these are necessary if all they're doing is returning the default label and value. They're optional functions for customisation (like adding images/icons.)

What we should do those is remove these and expose the optional functions to the consuming app by adding them to the props interface and hooking them up 👍